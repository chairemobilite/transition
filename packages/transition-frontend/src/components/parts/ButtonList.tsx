/*
 * Copyright 2022, Polytechnique Montreal and contributors
 *
 * This file is licensed under the MIT License.
 * License text available at https://opensource.org/licenses/MIT
 */
import React from 'react';

interface ButtonListProps {
    /**
     * Additional classes to give to this list
     */
    className?: string;
    children: React.ReactNode;
}

/**
 * A formatted list of buttons. Children can be Button elements
 * @param props
 * @returns
 */
const ButtonList: React.FunctionComponent<ButtonListProps> = (props: ButtonListProps) => {
    return <ul className={`_list-container${props.className ? ` ${props.className}` : ''}`}>{props.children}</ul>;
};

export default ButtonList;
