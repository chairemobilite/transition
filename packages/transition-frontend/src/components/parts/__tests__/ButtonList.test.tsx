/*
 * Copyright 2022, Polytechnique Montreal and contributors
 *
 * This file is licensed under the MIT License.
 * License text available at https://opensource.org/licenses/MIT
 */
import * as React from 'react';
import { create } from 'react-test-renderer';
import ButtonList from '../ButtonList';

test('Default props', () => {
    const input = create(<ButtonList
        key = 'key'
    >test</ButtonList>)
        .toJSON();
    expect(input).toMatchSnapshot();
});

test('With classname', () => {
    const input = create(<ButtonList
        key = 'key'
        className = 'myClass'
    >test</ButtonList>)
        .toJSON();
    expect(input).toMatchSnapshot();
});
