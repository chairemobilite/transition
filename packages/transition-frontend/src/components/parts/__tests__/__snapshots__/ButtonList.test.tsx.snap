// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default props 1`] = `
<ul
  className="_list-container"
>
  test
</ul>
`;

exports[`With classname 1`] = `
<ul
  className="_list-container myClass"
>
  test
</ul>
`;
